controller_manager/src/ros2_control_node.cpp

```diff
@@ -33,36 +68,43 @@ int main(int argc, char ** argv)
 
   auto cm = std::make_shared<controller_manager::ControllerManager>(executor, manager_node_name);
 
-  // TODO(anyone): Due to issues with the MutliThreadedExecutor, this control loop does not rely on
-  // the executor (see issue #260).
-  // When the MutliThreadedExecutor issues are fixed (ros2/rclcpp#1168), this loop should be
-  // converted back to a timer.
+  RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
+
   std::thread cm_thread(
     [cm]()
     {
-      RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
+      if (has_realtime_kernel())
+      {
+        if (!configure_sched_fifo(kSchedPriority))
+        {
+          RCLCPP_WARN(cm->get_logger(), "Could not enable FIFO RT scheduling policy");
+        }
+      }
+      else
+      {
+        RCLCPP_WARN(cm->get_logger(), "RT kernel was not detected");
+      }
 
-      rclcpp::Time current_time = cm->now();
-      rclcpp::Time previous_time = current_time;
-      rclcpp::Time end_period = current_time;
+      rclcpp::Time mut_previous_time = cm->now();
+      rclcpp::Time mut_end_period = mut_previous_time;
 
       // Use nanoseconds to avoid chrono's rounding
-      rclcpp::Duration period(std::chrono::nanoseconds(1000000000 / cm->get_update_rate()));
+      rclcpp::Duration period(std::chrono::nanoseconds(1'000'000'000 / cm->get_update_rate()));
 
       while (rclcpp::ok())
       {
         // wait until we hit the end of the period
-        end_period += period;
+        mut_end_period += period;
         std::this_thread::sleep_for(
-          std::chrono::nanoseconds((end_period - cm->now()).nanoseconds()));
+          std::chrono::nanoseconds((mut_end_period - cm->now()).nanoseconds()));
 
         // execute update loop
-        current_time = cm->now();
-        auto period = current_time - previous_time;
```

Comment:
I know it says that I shouldn't include "extra" changes. The aliasing of the variable period was driving me crazy. Also, cost stack variables make this easier to read than mutable stack variables that are never mutated.